pkg/reconcilers/targetref_reconciler.go

```diff
@@ -239,11 +232,110 @@ func (r *TargetRefReconciler) DeleteTargetBackReference(ctx context.Context,
 		delete(objAnnotations, annotationName)
 		targetObj.SetAnnotations(objAnnotations)
 		err := r.UpdateResource(ctx, targetObj)
-		logger.V(1).Info("DeleteTargetBackReference: update network resource",
-			"type", targetObjType, "name", targetObjKey, "err", err)
+		logger.V(1).Info("DeleteTargetBackReference: update network resource", "type", targetObjType, "name", targetObjKey, "err", err)
 		if err != nil {
 			return err
 		}
 	}
+
+	return nil
+}
+
+// Returns:
+// * list of gateways to which the policy applies for the first time
+// * list of gateways to which the policy no longer applies
+// * list of gateways to which the policy still applies
+func (r *TargetRefReconciler) ComputeGatewayDiffs(ctx context.Context, policy common.KuadrantPolicy, policyRefsConfig common.PolicyRefsConfig) (*GatewayDiff, error) {
+	logger, _ := logr.FromContext(ctx)
+
+	gwKeys, err := r.TargetedGatewayKeys(ctx, policy.GetTargetRef(), policy.GetNamespace())
+	if err != nil {
+		return nil, err
```

Comment:
I think we should only return an error here is it's other than not found.

Reply:
right. On NotFound error, it should return an empty list.

Reply:
I'll make the change.